app/code/community/Bolt/Boltpay/controllers/ShippingController.php

```diff
@@ -132,11 +134,12 @@ public function indexAction()
             // Check session cache for estimate.  If the shipping city or postcode, and the country code match,
             // then use the cached version.  Otherwise, we have to do another calculation
             ////////////////////////////////////////////////////////////////////////////////////////
-            $cached_address = unserialize(Mage::app()->getCache()->load("quote_location_".$quote->getId()));
+            $cacheKey = $this->generateCacheKey($quote->getId(), $quote->getShippingAddress(), $quote->getGrandTotal());
+            $cached_address = unserialize(Mage::app()->getCache()->load("quote_location_" . $cacheKey));
 
             if ($cached_address && ((strtoupper($cached_address["city"]) == strtoupper($address_data["city"])) || ($cached_address["postcode"] == $address_data["postcode"])) && ($cached_address["country_id"] == $address_data["country_id"])) {
                 //Mage::log('Using cached address: '.var_export($cached_address, true), null, 'shipping_and_tax.log');
-                $response = unserialize(Mage::app()->getCache()->load("quote_shipping_and_tax_estimate_".$quote->getId()));
+                $response = unserialize(Mage::app()->getCache()->load("quote_shipping_and_tax_estimate_" . $cacheKey));
             } else {
                 //Mage::log('Generating address from quote', null, 'shipping_and_tax.log');
                 //Mage::log('Live address: '.var_export($address_data, true), null, 'shipping_and_tax.log');
```

Comment:
can you add HTTP response header `X-Bolt-Cache-Hit: Miss`?

Reply:
added